SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1599,6 +1642,45 @@ bool ExtractComponentSetsFromSchemaJson(const FString& SchemaJsonPath, TMap<uint
 			SAFE_TRYGETFIELD((*CompObject), Number, "componentId", ComponentId);
 
 			ComponentMap.Add(ComponentName, ComponentId);
+
+			const TArray<TSharedPtr<FJsonValue>>* FieldArray;
+			SAFE_TRYGETFIELD((*CompObject), Array, "fields", FieldArray);
+
+			const uint32 FieldArrayIndex = OutFieldIdsArray.Num();
+			if (FieldArray->Num() > 0)
+			{
+				COND_SCHEMA_GEN_ERROR_AND_RETURN(OutComponentIdToFieldIdsIndex.Contains(ComponentId),
+												 TEXT("The schema bundle contains duplicate component IDs with component %s."),
+												 *ComponentName);
+				OutComponentIdToFieldIdsIndex.Add(ComponentId, FieldArrayIndex);
+				OutFieldIdsArray.AddDefaulted();
```

Comment:
nit: hold onto this array, like you do above, and wrap the below for loop in this if statement so it can still be accessed.

Reply:
thought about it, yeah, guess I can do that

Reply:
did that now